### PR TITLE
Adding Ziba's nested parameter dictionary changes, along with a test

### PR DESCRIPTION
**Pull request by irace**
Plus deleted old tests that weren't working anymore

@ziba77

### CODE REVIEW

File: Tests/TMSDKFunctionsTest.m [hunk 1]
```diff
@@ -62,4 +66,23 @@ - (void)testDictionaryToQueryWithRepeatedParameter {
...
+    NSDictionary *input = @{
+                            @"stringKey": @"value",
+                            @"arrayKey": @[@"arrayValue1", @"arrayValue2"],
+                            @"dictionaryKey": @{
+                                    @"innerDictionaryKey1": @"innerDictionaryValue1",
+                                    @"innerDictionaryKey2": @"innerDictionaryValue2"
+                                    },
+                            };
+    
+    NSString *expected = [NSString stringWithFormat:@"arrayKey=arrayValue1&arrayKey=arrayValue2&%@=innerDictionaryValue1&%@=innerDictionaryValue2&stringKey=value",
```
Comment (segiddins) [inline Tests/TMSDKFunctionsTest.m]: are those literal `%@`s intentional?

**Comment by segiddins** [comment]
Why are all those tests deleted?

**Comment by irace** [comment]
> Why are all those tests deleted?

Hadn’t been maintained and I didn't think they were really buying us much. They weren't testing logic just doing a bunch of mocking and then string matching.

This whole library is getting a pretty dramatic overhaul shortly.

File: Tests/TMSDKFunctionsTest.m (hunk 1 above)
Comment (irace) [inline Tests/TMSDKFunctionsTest.m]: Yes

**Pull request merged**